frontend/src/lib/k8s/node.ts

```diff
@@ -71,6 +109,50 @@ class Node extends makeKubeObject<KubeNode>('node') {
   getInternalIP(): string {
     return this.status.addresses.find(address => address.type === 'InternalIP')?.address || '';
   }
+
+  async shell(onExec: StreamResultsCb) {
+    const cluster = getCluster();
+    if (!cluster) {
+      return {};
+    }
+
+    const clusterSettings = helpers.loadClusterSettings(cluster);
+    let image = clusterSettings.dropShellLinuxImage;
+    const podName = `node-shell-${this.getName()}-${uniqueString()}`;
+    if (image === null || image === '') {
+      image = DEFAULT_DROP_SHELL_LINUX_IMAGE;
+    }
+    console.log(image);
+    const kubePod = shellPod(podName, this.getName(), image!!);
+    await apply(kubePod);
+    const command = [
+      'sh',
+      '-c',
+      '((clear && bash) || (clear && zsh) || (clear && ash) || (clear && sh))',
+    ];
+    const tty = true;
+    const stdin = true;
+    const stdout = true;
+    const stderr = true;
+    const commandStr = command.map(item => '&command=' + encodeURIComponent(item)).join('');
+    const url = `/api/v1/namespaces/kube-system/pods/${podName}/exec?container=shell${commandStr}&stdin=${
+      stdin ? 1 : 0
+    }&stderr=${stderr ? 1 : 0}&stdout=${stdout ? 1 : 0}&tty=${tty ? 1 : 0}`;
```

Comment:
The 1st time I tried your PR, it failed to show any terminal. I think it's because the image was not pulled and it took longer? After re-trying it did work. Maybe you can reproduce it by using an always pull policy in the pod during dev.

I wonder if we can somehow show a spinner and wait till the pod is available, for a better UX.